src/inference/hmc.jl

```diff
@@ -354,12 +398,23 @@ Arguments:
 - `init_ϵ::Float64` : Inital step size; 0 means automatically searching using a heuristic procedure.
 
 """
-struct NUTS{AD,space,metricT<:AHMC.AbstractMetric} <: AdaptiveHamiltonian{AD}
+struct NUTS{
+    AD,
+    space,
+    metricT <: AHMC.AbstractMetric,
+    TS<:AHMC.AbstractTrajectorySampler,
+    TC<:AHMC.AbstractTerminationCriterion,
+    I<:AHMC.AbstractIntegrator,
+    A<:AHMC.AbstractAdaptor
```

Comment:
Is it sufficient to make these type parameters without corresponding fields (apart from `I`)? I.e., do the types contain all information or do we need values?